tests/Unit/SDK/Metrics/ValueRecorderTest.php

```diff
@@ -5,149 +5,162 @@
     {
-        $metric = new ValueRecorder('name', 'description');
-        $metric->record(5);
-        $this->assertEquals(1, $metric->getCount());
-        $this->assertEquals(5, $metric->getMax());
-        $this->assertEquals(5, $metric->getMin());
-        $this->assertEquals(5, $metric->getSum());
-        $metric->record(-2);
-        $this->assertEquals(2, $metric->getCount());
-        $this->assertEquals(5, $metric->getMax());
-        $this->assertEquals(-2, $metric->getMin());
-        $this->assertEquals(3, $metric->getSum());
+        $this->metric->record(5);
+        $this->assertEquals(1, $this->metric->getCount());
+        $this->assertEquals(5, $this->metric->getMax());
+        $this->assertEquals(5, $this->metric->getMin());
+        $this->assertEquals(5, $this->metric->getSum());
+        $this->metric->record(-2);
+        $this->assertEquals(2, $this->metric->getCount());
+        $this->assertEquals(5, $this->metric->getMax());
+        $this->assertEquals(-2, $this->metric->getMin());
+        $this->assertEquals(3, $this->metric->getSum());
     }
 
     public function test_valid_negative_and_positive_record(): void
     {
-        $metric = new ValueRecorder('name', 'description');
-        $metric->record(-5);
-        $this->assertEquals(1, $metric->getCount());
-        $this->assertEquals(-5, $metric->getMax());
-        $this->assertEquals(-5, $metric->getMin());
-        $this->assertEquals(-5, $metric->getSum());
-        $metric->record(2);
-        $this->assertEquals(2, $metric->getCount());
-        $this->assertEquals(2, $metric->getMax());
-        $this->assertEquals(-5, $metric->getMin());
-        $this->assertEquals(-3, $metric->getSum());
+        $this->metric->record(-5);
+        $this->assertEquals(1, $this->metric->getCount());
+        $this->assertEquals(-5, $this->metric->getMax());
+        $this->assertEquals(-5, $this->metric->getMin());
+        $this->assertEquals(-5, $this->metric->getSum());
+        $this->metric->record(2);
+        $this->assertEquals(2, $this->metric->getCount());
+        $this->assertEquals(2, $this->metric->getMax());
+        $this->assertEquals(-5, $this->metric->getMin());
+        $this->assertEquals(-3, $this->metric->getSum());
     }
 
     public function test_valid_positive_float_record(): void
     {
-        $metric = new ValueRecorder('name', 'description');
-        $metric->record(5.2222);
-        $this->assertEquals(1, $metric->getCount());
-        $this->assertEquals(5.2222, $metric->getMax());
-        $this->assertEquals(5.2222, $metric->getMin());
-        $this->assertEquals(5.2222, $metric->getSum());
-        $metric->record(2.6666);
-        $this->assertEquals(2, $metric->getCount());
-        $this->assertEquals(5.2222, $metric->getMax());
-        $this->assertEquals(2.6666, $metric->getMin());
-        $this->assertEquals(7.8888, $metric->getSum());
+        $this->metric->record(5.2222);
```

Comment:
The tested metthod (`record`) could simply expect an float and all the above tests would not be needed, as integers are floats as well.

Reply:
done - added a `float` type-hint, and removed some casting and checks that are no longer needed